test/extended/builds/pipeline.go

```diff
@@ -192,918 +317,663 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By(fmt.Sprintf("calling oc new-app -f %q", podTemplateSlavePipelinePath))
-			err = oc.Run("new-app").Args("-f", podTemplateSlavePipelinePath).Execute()
-			o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("verify credential created since label should be there")
+				// NOTE, for the credential URL in Jenkins
+				// it returns rc 200 with no output if credential exists and a 404 if it does not exists
+				_, err = j.WaitForContent("", 200, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("starting the pipeline build and waiting for it to complete")
-			br, err := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
-			if err != nil || !br.BuildSuccess {
-				exutil.DumpBuilds(oc)
-			}
-			debugAnyJenkinsFailure(br, oc.Namespace()+"-openshift-jee-sample", oc, true)
-			br.AssertSuccess()
+				g.By("verify credential deleted when label removed")
+				err = oc.Run("label").Args("secret", secretName, secretCredentialSyncLabel+"-").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = j.WaitForContent("", 404, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("getting job log, making sure job log has success message")
-			out, err := j.GetJobConsoleLogsAndMatchViaBuildResult(br, "Finished: SUCCESS")
-			o.Expect(err).NotTo(o.HaveOccurred())
-			g.By("making sure job log ran with our config map slave pod template")
-			o.Expect(out).To(o.ContainSubstring("Running on slave-jenkins"))
-		})
-	})
+				g.By("verify credential added when label added")
+				err = oc.Run("label").Args("secret", secretName, secretCredentialSyncLabel+"=true").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = j.WaitForContent("", 200, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-	g.Context("Pipeline using jenkins-client-plugin", func() {
-		g.BeforeEach(func() {
-			setupJenkins(jenkinsEphemeralTemplatePath)
+				g.By("verify credential deleted when secret deleted")
+				err = oc.Run("delete").Args("secret", secretName).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = j.WaitForContent("", 404, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
-				ticker = jenkins.StartJenkinsMemoryTracking(oc, oc.Namespace())
-			}
+				// no need to clean up, last operation above deleted the secret
+			})
 
-			g.By("waiting for builder service account")
-			err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
-			o.Expect(err).NotTo(o.HaveOccurred())
-		})
+			g.By("Pipeline using config map slave")
```

Comment:
at some point in the future (ie not in this PR) i think we could speed up some of these tests by just checking that the jenkins configuration is correct (has the slave config defined) instead of actually running a pipeline w/ the slave configuration.  Looks like we have 2-3 tests that really could be boiled down to "is the config right" instead of "can we fully run a pipeline with the config"

Reply:
generally agreed ... though one small clarification which I mentioned in scrum: that while we may not need 2 or 3 tests that run pipelines leveraging the agent image, as long as we have at least one of these tests somewhere, in any of these scenarios 

I'll add a TODO comment that attempts to capture this

Reply:
yup, agreed that we do need one that's actually exercising the main thing (e.g. building a maven project)